Mathlib/Data/BitVec/Lemmas.lean

```diff
@@ -218,4 +244,62 @@ instance : CommSemiring (BitVec w) :=
     toFin_zero toFin_one toFin_add toFin_mul (Function.swap toFin_nsmul)
     toFin_pow toFin_natCast
 
+/-!
+## Extensionality
+-/
+
+/-- If two bitvectors agree on all in-bound bits, then they agree on all bits -/
+theorem getLsb_eq_of_getLsb' {x y : BitVec w} (h : ∀ (i : Fin w), x.getLsb' i = y.getLsb' i) :
+    ∀ (i : ℕ), x.getLsb i = y.getLsb i := by
+  simp only [getLsb, testBit]
+  intro i
+  cases' lt_or_le i w with hi hi
+  · exact h ⟨i, hi⟩
+  · have (z : BitVec w) : z.toNat < 2 ^ i :=
+      Nat.lt_of_lt_of_le z.toNat_lt (pow_le_pow (le_succ 1) hi)
+    rw [Nat.shiftRight_eq_zero_of_lt (this x), Nat.shiftRight_eq_zero_of_lt (this y)]
+
+/-- If two bitvectors agree on all bits, then they are equal. See also `Std.BitVec.ext_msb` -/
+@[ext]
+theorem ext_lsb {x y : BitVec w} (h : ∀ i, x.getLsb' i = y.getLsb' i) : x = y := by
+  apply toNat_inj.mp
+  apply Nat.eq_of_testBit_eq
+  simp only [testBit_toNat]
+  exact getLsb_eq_of_getLsb' h
+
+theorem getMsb'_eq_getLsb' (x : BitVec w) (i : Fin w) :
+    x.getMsb' i = x.getLsb' i.rev := by
```

Comment:
```suggestion
theorem getLsb'_rev (x : BitVec w) (i : Fin w) :
    x.getLsb' i.rev = x.getMsb' i := by
```
and then probably we should have a `getMsb'_rev` lemma too.

Reply:
A added a placeholder for getMsb'_rev and marked this as waiting-for-author.